host/src/lib.rs

```diff
@@ -425,6 +464,10 @@ impl ScalarUDFImpl for WasmScalarUdf {
     }
 
     fn return_type(&self, arg_types: &[DataType]) -> DataFusionResult<DataType> {
+        if let Some(return_type) = &self.return_type {
+            return Ok(return_type.clone());
+        }
+
```

Comment:
After #157 is merged, we should probably emit a debug log in case the return type was NOT cached.